src/headers.js

```diff
@@ -35,68 +44,103 @@ function fill (headers, object) {
 
 class Headers {
   constructor (init) {
-    this[kHeaders] = new Map()
+    this[kHeaders] = []
 
     if (init && typeof init === 'object') {
       fill(this, init)
     }
   }
 
   append (name, value) {
-    const [normalizedHeaderName, normalizedHeaderValue] = normalizeAndValidateHeaderValue(name, value)
-
-    const existing = this[kHeaders].get(normalizedHeaderName)
-    const newHeaderValue = existing ? existing + ', ' + normalizedHeaderValue : normalizedHeaderValue
-    this[kHeaders].set(normalizedHeaderName, newHeaderValue)
+    const [normalizedName, normalizedValue] = normalizeAndValidateHeaderValue(name, value)
+
+    let index = this[kHeaders].length
+    for (let i = 0; i < this[kHeaders].length; i += 2) {
+      if (normalizedName === this[kHeaders][i]) {
+        this[kHeaders][i + 1] += `, ${normalizedValue}`
+        return
+      } else if (this[kHeaders][i] > normalizedName) {
```

Comment:
Doesn’t this require localeCompare?

Reply:
maybe, I think the normalization and validation should take care of that, but its a good point. I'll look into it

Reply:
Okay fetch spec says that when sorting, you are supposed to compare using [byte less than](https://infra.spec.whatwg.org/#byte-less-than) which IIRC is all based on ASCII byte values? So honestly I do not know 😅 might have to wait until I port over the node-fetch tests and see if they cover that case

Reply:
I'm resolving this for now and opening an investigate issue for the future

Reply:
https://github.com/Ethan-Arrowood/undici-fetch/issues/39